File: 2020/erik/Day01/src/Day01.jl
module Day01

using DelimitedFiles

function readdata(path)
    readdlm(path, '\t', Int)
end

function part1(numbers)
    reduce(*, filter(n -> in(2020 - n, numbers), numbers))
end

function part2(numbers)
    for m in numbers
        pair = filter(n -> in(2020 - m - n, numbers), numbers)
        if length(pair) == 2
            return m * pair[1] * pair[2]
        end
    end
end

function main()
    data = readdata("input/input01.txt")
    println("> Part one: ", part1(data))
    println("> Part two: ", part2(data))
end

end # module

File: archive/2021/torarvid/src/day01.rs
use crate::util::file_by_lines_as_i32;

pub fn run() {
    let numbers = file_by_lines_as_i32("day01.txt");

    let part1 = numbers.windows(2).filter(|w| w[0] < w[1]).count();
    println!("Part 1: {}", part1);

    // Part 2
    let grouped: Vec<_> = numbers.windows(3).map(|w| w[0] + w[1] + w[2]).collect();

    let part2 = grouped.windows(2).filter(|w| w[0] < w[1]).count();
    println!("Part 2: {}", part2);
}
